src/OpenTelemetry.Instrumentation.SqlClient/README.md

```diff
@@ -221,6 +221,34 @@ using var tracerProvider = Sdk.CreateTracerProviderBuilder()
     .Build();
 ```
 
+## Filter
+
+This option allows to filter out activities based on the properties of the
+`SqlCommand` object being instrumented using a `Func<object, bool>`.
+The function receives an instance of the raw `SqlCommand` and should return
+`true` if the telemetry is to be collected, and `false` if it should not.
+The example below filters out all commands that are not stored procedures.
+
```

Comment:
We should probably mention that one should always cast the `object` to `Microsoft.Data.SqlClient.Command` when using the Filter option.

Reply:
updated with text mentioning both `System.Data.SqlClient.SqlCommand` and `Microsoft.Data.SqlClient.SqlCommand` as - according to my understanding - it can be either one of those, hence why object is used as the parameter type.